feeder/main.py

```diff
@@ -34,26 +23,49 @@ def handle_exception(loop, context):
         logging.error("Caught global exception: %s", msg)
 
 
+frontend = Path("./static/build/index.html")
 frontend_template = Jinja2Templates(directory="static/build")
-async_loop = asyncio.get_event_loop()
-async_loop.set_exception_handler(handle_exception)
-client = FeederClient()
-broker = FeederBroker()
 
-mqtt_enabled_routers = [feeder, pet]
-for mqtt_router in mqtt_enabled_routers:
-    mqtt_router.router.client = client
-    mqtt_router.router.broker = broker
+
+async def render_frontend(full_path: str, request: Request):
+    frontend_paths = ["settings", "feeders", ""]
+    if full_path not in frontend_paths:
+        raise HTTPException(status_code=404)
+
+    if frontend.exists():
+        build_path = "build"
+        if settings.app_root:
+            build_path = f"{settings.app_root[1:]}/build"
+        return frontend_template.TemplateResponse(
+            "index.html",
+            {
+                "request": request,
+                "build_path": build_path,
+                "root_path": settings.app_root,
+            },
+        )
+
+    logger.warning("Caught frontend request without built frontend.")
+    raise HTTPException(status_code=500, detail="Frontend not built!")
 
 
 def get_application() -> FastAPI:
```

Comment:
Maybe this should be named `create_application`? The naming currently makes it seem like it's idempotent.